tests/functional/TestSmokeTest/tests/timeout_test.go

```diff
@@ -0,0 +1,126 @@
+	nodelist := make([]string, nodes)
+
+	for index, ip := 0, 100; index < nodes; index, ip = index+1, ip+1 {
+		nodelist[index] = "192.168.10." + fmt.Sprintf("%v", ip)
+	}
+
+	return nodelist
+}
+
+func stopSleep(node string, exec *ssh.SshExec) error {
+
+	cmd := []string{" pkill -9 sleep || :"}
+	_, err := exec.ConnectAndExec(node+":"+SSHPort, cmd, 10, true)
+	return err
+}
+func addPostScript(timeOut int, node string, exec *ssh.SshExec) error {
+
+	//create and change file permission pre-script file
+	createFile := fmt.Sprintf("install -m 777 /dev/null %s", preScriptFile)
+
+	//write script which sleeps for minutes
+	dumpData := fmt.Sprintf("echo \"#! /bin/bash\n sleep %dm \">%s", timeOut, preScriptFile)
+
+	cmd := []string{createFile, dumpData}
+	_, err := exec.ConnectAndExec(node+":"+SSHPort, cmd, 10, true)
+	return err
+}
+
+func deletePreScript(node string, exec *ssh.SshExec) error {
+
+	//delete pre script file
+	removeFile := fmt.Sprintf("rm -rf %s", preScriptFile)
+	cmd := []string{removeFile}
+	_, err := exec.ConnectAndExec(node+":"+SSHPort, cmd, 10, true)
+	return err
+}
+
+func removeTimeout(sleep uint64, exec *ssh.SshExec, t *testing.T) {
+	time.Sleep(time.Duration(sleep) * time.Minute)
+	for _, node := range getNodes() {
+		err := deletePreScript(node, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		err = stopSleep(node, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	}
+
+}
+func TestGlusterTimeout(t *testing.T) {
+
+	// Setup the VM storage topology
+	teardownCluster(t)
+	setupCluster(t, 4, 8)
+	defer teardownCluster(t)
+
+	na := testutils.RequireNodeAccess(t)
+	//add postscript in all gluster nodes
+	exec := na.Use(logger)
+	for _, node := range getNodes() {
+		err := addPostScript(6, node, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+
+	//removeTimeout deletes the postscript file and stops the sleep process
+	//which was started as part of pre script
+	//sleep for 3 minutes to validate we are not getting any timeout error
+	//from gluster cli
+	go removeTimeout(3, exec, t)
+
+	//Create a volume
+	volReq := &api.VolumeCreateRequest{}
+	volReq.Size = 1
+	volReq.Snapshot.Enable = true
+	volReq.Snapshot.Factor = 1.5
+	volReq.Durability.Type = api.DurabilityReplicate
+
+	simplevol, err := heketi.VolumeCreate(volReq)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	// Delete volume
+	err = heketi.VolumeDelete(simplevol.Id)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	for _, node := range getNodes() {
+		//add post script with sleep of 12 minutes
+		err := addPostScript(12, node, exec)
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	}
+	//volume creation should fail as we have the sleep of 12 minutes in pre scripts
+	//which causes the gluster cli to timeout for 10 minutes
+	_, err = heketi.VolumeCreate(volReq)
+	tests.Assert(t, err != nil, "expected err != nil, got:", err)
```

Comment:
In the failure case, `err == nil` does not give you useful output, no need to log `err` :)